### PR TITLE
Move config template and snap 

### PR DESCRIPTION
**Pull request by sithlord48**
## Contributor Checklist:

* [ ] This is a user-visible change and I have created a file in the `doc/newsfragments` directory (and made sure to read the `README.md` in that directory)
* [X] This is not a user-visible change

 - Move `res/config.h.in` -> `src/lib/config.h.in`
   - Move the configure_file for it from `CMakeLists.txt` -> `src/lib/CMakeLists.txt`
   - Use `@Only` to configure the file. to be consistent with our other cmake template files.
 - Move `snap` folder into `dist`, with the other distribution files.
   - Update `snapcraft.yml`  to use the proper appid `io.github.input_leap.InputLeap`

### CODE REVIEW

File: dist/snap/snapcraft.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ apps:
       # Fallback to XWayland if running in a Wayland session.
       DISABLE_WAYLAND: 1
     desktop: usr/share/applications/input-leap.desktop
-    common-id: com.github.input-leap.input-leap
+    common-id: io.github.input_leap.InputLeap
```
Comment (p12tic) [inline dist/snap/snapcraft.yaml:18]: Why the previous value isn't good? We do not own input_leap.github.io or input_leap.github.com.

**Review by p12tic — CHANGES_REQUESTED** [review]
One comment, good otherwise.

**Comment by p12tic** [comment]
@sithlord48 I have directed you multiple times about the format of commit messages that I prefer. Are you trolling me? :)

File: dist/snap/snapcraft.yaml (hunk 1 above)
Comment (sithlord48) [reply dist/snap/snapcraft.yaml:18]: because the previous value is not what appstream has agreed upon

Comment (sithlord48) [reply dist/snap/snapcraft.yaml:18]: >"Each AppStream component has a unique identifier (the value of the <id/> tag), which is unique across all distributions" 
 our name for flathub has to be `io.github.input_leap.InputLeap`  so for snap it must also

**Comment by sithlord48** [comment]
> @sithlord48 I have directed you multiple times about the format of commit messages that I prefer. Are you trolling me? :)

what is wrong with my commit messages?

**Comment by p12tic** [comment]
> what is wrong with my commit messages?

We aren't using conventional commits, thus `fix:`, `release:`, etc are redundant.

**Review by p12tic — APPROVED** [review]

**Pull request merged**